README.md
# Force WP Template for The Events Calendar Plugin

changelog.txt
EventsCalendarEnforcer v1.0.0
October 29, 2020
- Added The Events Calendar Plugin tribe_events_views_v2_use_wp_template_hierarchy filter, returns true.

gulpfile.js
/**
 * Gulpfile.
 *
 * Gulp with WordPress.
 *
 * Implements:
 *      1. Live reloads browser with BrowserSync.
 *      2. CSS: Sass to CSS conversion, error catching, Autoprefixing, Sourcemaps,
 *         CSS minification, and Merge Media Queries.
 *      3. JS: Concatenates & uglifies Vendor and Custom JS files.
 *      4. Images: Minifies PNG, JPEG, GIF and SVG images.
 *      5. Watches files for changes in CSS or JS.
 *      6. Watches files for changes in PHP.
 *      7. Corrects the line endings.
 *      8. InjectCSS instead of browser page reload.
 *      9. Generates .pot file for i18n and l10n.
 *
 * @tutorial https://github.com/ahmadawais/WPGulp
 * @author Ahmad Awais <https://twitter.com/MrAhmadAwais/>
 */

/**
 * Load WPGulp Configuration.
 *
 * TODO: Customize your project in the wpgulp.js file.
 */
const config = require( './wpgulp.config.js' );

/**
 * Load Plugins.
 *
 * Load gulp plugins and passing them semantic names.
 */
const gulp = require( 'gulp' ); // Gulp of-course.
const parameterized = require('gulp-parameterized');

// CSS related plugins.
const sass = require( 'gulp-sass' ); // Gulp plugin for Sass compilation.

// JS related plugins.
const uglify = require('gulp-uglify-es').default; // Minifies JS files.

// HTML related plugins
const htmlbeautify = require('gulp-html-beautify'); // Beautify HTML/PHP files

// Image related plugins.
const imagemin = require( 'gulp-imagemin' ); // Minify PNG, JPEG, GIF and SVG images with imagemin.

// Utility related plugins.
const concat = require( 'gulp-concat' ); // Concatenates files.
const lineec = require( 'gulp-line-ending-corrector' ); // Consistent Line Endings for non UNIX systems. Gulp Plugin for Line Ending Corrector (A utility that makes sure your files have consistent line endings).
const notify = require( 'gulp-notify' ); // Sends message notification to you.

const fs = require('fs'); // File System


events-calendar-enforcer.php
<?php
/**
 * Plugin Name: Force WP Template for The Events Calendar Plugin
 * Plugin URI: "https://github.com/Danny-Guzman/events-calendar-enforcer"
 * Description: Forces The Events Calendar Plugin to use the WordPress template hierarchy for events instead of its own templates.
 * Author: Danny Guzman
 * Version: 1.0.0
 * Author URI: "https://github.com/Danny-Guzman/"
 *
 * @package TribeEventsEnforcer
 */

define( 'TRIBE_EVENTS_ENFORCER_URL', plugin_dir_url( __FILE__ ) );
define( 'TRIBE_EVENTS_ENFORCER_DIR', plugin_dir_path( __FILE__ ) );

add_action( 'admin_init', 'tribe_events_enforcer_admin_init' );
add_filter( 'tribe_events_views_v2_use_wp_template_hierarchy', 'tribe_events_enforcer_force_template', 10, 4 );

/**
 * Admin Init
 *
 * Triggered before any other hook when a user accesses the admin area.
 * Note, this does not just run on user-facing admin screens.
 * It runs on admin-ajax.php and admin-post.php as well.
 *
 * @link https://codex.wordpress.org/Plugin_API/Action_Reference/admin_init
 * @return void
 */
function tribe_events_enforcer_admin_init() {
	require_once TRIBE_EVENTS_ENFORCER_DIR . 'core/class-tribe-events-enforcer-plugin-update.php';
}

/**
 * With this filter, we can tell The Events Calendar to use the WordPress template hierarchy for events instead of its own templates.
 *
 * @link https://theeventscalendar.com/knowledgebase/k/using-the-template-management-filters/
 *
 * @param  boolean $hijack Whether to transitions the calendar from plugin templates to theme templates.
 * @param  mixed   $template The template located by WordPress.
 * @param  mixed   $context The immutable, global object instance that serves as the Tribe template context.
 * @param  mixed   $query The global $wp_query.
 * @return boolean
 */
function tribe_events_enforcer_force_template( $hijack, $template, $context, $query ) {
	return true;
}


core/class-tribe-events-enforcer-plugin-update.php
<?php
/**
 * Plugin Updater
 *
 * @see https://github.com/WordPress/WordPress/blob/master/wp-admin/update.php
 * @see https://github.com/WordPress/WordPress/blob/master/wp-admin/includes/class-theme-upgrader.php
 * @see https://github.com/WordPress/WordPress/blob/master/wp-admin/includes/class-wp-upgrader.php
 *
 * @package TribeEventsEnforcer
 */

if ( ! class_exists( 'Tribe_Events_Enforcer_Plugin_Update' ) ) {
	/**
	 * TribeEventsEnforcer Plugin Upgrader
	 */
	class Tribe_Events_Enforcer_Plugin_Update {

		/**
		 * Member Variable
		 *
		 * @var array $plugin_name Plugin name.
		 */
		protected $plugin_name;
		/**
		 * Member Variable
		 *
		 * @var string $current_version Current Plugin Version.
		 */
		protected $current_version;

		/**
		 * Member Variable
		 *
		 * @var string $slug Plugin Slug name.
		 */
		protected $slug;

		/**
		 * Member Variable
		 *
		 * @var string $plugin_file Plugin filename.
		 */
		protected $plugin_file;

		/**
		 * Member Variable
		 *
		 * @var string $transient_name Name of update transient.
		 */
		protected $transient_name = 'tribe_events_enforcer_update_plugins';

		/**
		 * Member Variable
		 *
		 * @var string $repo Plugin repo location.
		 */
		protected $repo = 'https://api.github.com/repos/Danny-Guzman/events-calendar-enforcer';

		/**
		 * Member Variable
		 *
		 * @var array $args Contains Header arguments used during the update process.
		 */
		protected $args;

		/**
		 * Initialize a new instance of the WordPress Auto-Update class
		 *
		 * @param string $plugin_slug Plugin slug name.
		 */
		public function __construct( $plugin_slug ) {
			$plugin_data = get_plugin_data( sprintf( '%1$s/%2$s/%2$s.php', WP_PLUGIN_DIR, $plugin_slug ) );

			// Set the class public variables.
			$this->current_version = $plugin_data['Version'];
			$this->plugin_name     = $plugin_data['Name'];
			$this->slug            = $plugin_slug;
			$this->plugin_file     = sprintf( '%1$s/%1$s.php', $plugin_slug );

			$this->args = array(
				'headers' => array(
					'Accept:'       => 'application/vnd.github.v3+json',
					'application/vnd.github.VERSION.raw',
					'application/octet-stream',
				),
			);

			// define the alternative API for plugin update checking.
			add_filter( 'pre_set_site_transient_update_plugins', array( $this, 'tribe_events_enforcer_check_plugin_update' ) );

			// Define the alternative response for information checking.
			add_filter( 'site_transient_update_plugins', array( $this, 'tribe_events_enforcer_add_plugins_to_update_notification' ) );

			add_filter( 'plugins_api', array( $this, 'tribe_events_enforcer_update_plugins_changelog' ), 20, 3 );

			// Define the alternative response for download_package which gets called during theme upgrade.
			add_filter( 'upgrader_pre_download', array( $this, 'download_package' ), 10, 3 );

			// Define the alternative response for upgrader_pre_install.
			add_filter( 'upgrader_source_selection', array( $this, 'tribe_events_enforcer_upgrader_source_selection' ), 10, 4 );

		}

		/**
		 * Alternative theme download for the WordPress Updater
		 *
		 * @see https://github.com/WordPress/WordPress/blob/master/wp-admin/includes/class-wp-upgrader.php#L265
		 *
		 * @param  bool        $reply Whether to bail without returning the package. Default false.
		 * @param  string      $package The package file name.
		 * @param  WP_Upgrader $upgrader The WP_Upgrader instance.
		 *
		 * @return string
		 */
		public function download_package( $reply, $package, $upgrader ) {
			if ( isset( $upgrader->skin->plugin_info ) && $upgrader->skin->plugin_info['Name'] === $this->plugin_name ) {
				$theme = wp_remote_retrieve_body( wp_remote_get( $package, array_merge( $this->args, array( 'timeout' => 60 ) ) ) );
				// Now use the standard PHP file functions.
				$fp = fopen( sprintf( '%1$s/%2$s.zip', plugin_dir_path( __DIR__ ), $this->slug ), 'w' );
				fwrite( $fp, $theme );
				fclose( $fp );

				return sprintf( '%1$s/%2$s.zip', plugin_dir_path( __DIR__ ), $this->slug );
			}
			return $reply;
		}

		/**
		 * Alternative API for checking for plugin updates.
		 *
		 * @param  array $update_transient Transient containing plugin updates.
		 *
		 * @return array
		 */
		public function tribe_events_enforcer_check_plugin_update( $update_transient ) {
			if ( ! isset( $update_transient->checked ) ) {
				return $update_transient;
			}

			$plugins = $update_transient->checked;

			$last_update = new stdClass();

			$payload = wp_remote_get( sprintf( '%1$s/releases/latest', $this->repo ), $this->args );

			if ( ! is_wp_error( $payload ) && wp_remote_retrieve_response_code( $payload ) === 200 ) {
				$payload = json_decode( wp_remote_retrieve_body( $payload ) );

				if ( ! empty( $payload ) && version_compare( $payload->tag_name, $this->current_version, '>' ) ) {
					$obj                 = new StdClass();
					$obj->name           = $this->plugin_name;
					$obj->slug           = $this->slug;
					$obj->plugin         = $this->plugin_file;
					$obj->new_version    = $payload->tag_name;
					$obj->published_date = ( new DateTime( $payload->published_at ) )->format( 'm/d/Y' );
					$obj->package        = $payload->zipball_url;
					$obj->tested         = '5.4.1';

					$theme_response = array( $this->plugin_file => $obj );

					$update_transient->response = array_merge( ! empty( $update_transient->response ) ? $update_transient->response : array(), $theme_response );

					$last_update->checked  = $plugins;
					$last_update->response = $theme_response;
				} else {
					delete_site_transient( $this->transient_name );
				}
			}

			$last_update->last_checked = time();
			set_site_transient( $this->transient_name, $last_update );

			return $update_transient;
		}

		/**
		 * Adds the EventsCalendarEnforce Plugin Update Notification to List of Available Updates.
		 *
		 * @param  array $update_transient Transient containing plugin updates.
		 *
		 * @return array
		 */
		public function tribe_events_enforcer_add_plugins_to_update_notification( $update_transient ) {
			$tribe_events_enforcer_update_plugins = get_site_transient( $this->transient_name );
			if ( ! is_object( $tribe_events_enforcer_update_plugins ) || ! isset( $tribe_events_enforcer_update_plugins->response ) ) {
				return $update_transient;
			}
			// Fix for warning messages on Dashboard / Updates page.
			if ( ! is_object( $update_transient ) ) {
				$update_transient = new stdClass();
			}

			$update_transient->response = array_merge(
				! empty( $update_transient->response ) ? $update_transient->response : array(),
				$tribe_events_enforcer_update_plugins->response
			);

			return $update_transient;
		}

		/**
		 * Filters the response for the current WordPress.org Plugin Installation API request.
		 *
		 * @see https://developer.wordpress.org/reference/functions/plugins_api/
		 *
		 * @param  false|object|array $result The result object or array. Default false.
		 * @param  string             $action The type of information being requested from the Plugin Installation API.
		 * @param  object             $args Plugin API arguments.
		 *
		 * @return false|object|array
		 */
		public function tribe_events_enforcer_update_plugins_changelog( $result, $action, $args ) {
			if ( isset( $args->slug ) && $args->slug === $this->slug ) {
				$tribe_events_enforcer_update_plugins = get_site_transient( $this->transient_name );
				if ( isset( $tribe_events_enforcer_update_plugins->response ) && isset( $tribe_events_enforcer_update_plugins->response[ $this->plugin_file ] ) ) {
					$tmp = $this->plugin_details();

					$tmp['version']      = $tribe_events_enforcer_update_plugins->response[ $this->plugin_file ]->new_version;
					$tmp['last_updated'] = $tribe_events_enforcer_update_plugins->response[ $this->plugin_file ]->published_date;

					$tmp['sections']['Changelog'] = $this->tribe_events_enforcer_get_plugin_changelog( $tmp['version'] );

					return (object) $tmp;
				}
			}
			return $result;
		}

		/**
		 * Retrieve Plugin Changelog from repository
		 *
		 * @param  string $ver Repository branch.
		 *
		 * @return string
		 */
		public function tribe_events_enforcer_get_plugin_changelog( $ver = 'master' ) {
			$logurl = sprintf( '%1$s/contents/changelog.txt?ref=%2$s', $this->repo, $ver );

			$changelog = wp_remote_get( $logurl, $this->args );

			if ( ! is_wp_error( $changelog ) && 200 === wp_remote_retrieve_response_code( $changelog ) ) {
				return sprintf( '<pre style="white-space: pre-line !important;">%1$s</pre>', base64_decode( json_decode( wp_remote_retrieve_body( $changelog ) )->content ) );
			} else {
				return 'No Changelog Available';
			}
		}

		/**
		 * Filters the source file location for the upgrade package.
		 *
		 * @see https://github.com/WordPress/WordPress/blob/master/wp-admin/includes/class-wp-upgrader.php#L524
		 *
		 * @param  string      $src File source location.
		 * @param  string      $rm_src Remote file source location.
		 * @param  WP_Upgrader $upgr WP_Upgrader instance.
		 * @param  array       $options Extra arguments passed to hooked filters.
		 *
		 * @return string
		 */
		public function tribe_events_enforcer_upgrader_source_selection( $src, $rm_src, $upgr, $options ) {

			if ( ! isset( $options['plugin'] ) || $options['plugin'] !== $this->plugin_file ) {
				return $src;
			}

			$tmp = explode( '/', $src );
			array_shift( $tmp );
			array_pop( $tmp );
			$tmp[ count( $tmp ) - 1 ] = $tmp[ count( $tmp ) - 2 ];
			$tmp                      = sprintf( '/%1$s/', implode( '/', $tmp ) );

			rename( $src, $tmp );

			return $tmp;
		}

		/**
		 * Plugin Details
		 *
		 * @see https://developer.wordpress.org/reference/functions/plugins_api/
		 * @return array
		 */
		public function plugin_details() {
			$view_details = array(
				'slug'     => plugin_basename( plugin_dir_path( __DIR__ ) ),
				'author'   => 'Danny Guzman',
				'name'     => 'Force WP Template for The Events Calendar Plugin',
				'sections' => array(
					'Description' => 'Forces The Events Calendar Plugin to use the WordPress template hierarchy for events instead of its own templates.',
				),
				'requires' => '5.0.0',
				'tested'   => '5.4.1',
			);

			return $view_details;
		}

	}
}

new Tribe_Events_Enforcer_Plugin_Update( plugin_basename( plugin_dir_path( __DIR__ ) ) );



core/class-volunteer-match-plugin-update.php
<?php
/**
 * Plugin Updater
 *
 * @see https://github.com/WordPress/WordPress/blob/master/wp-admin/update.php
 * @see https://github.com/WordPress/WordPress/blob/master/wp-admin/includes/class-theme-upgrader.php
 * @see https://github.com/WordPress/WordPress/blob/master/wp-admin/includes/class-wp-upgrader.php
 *
 * @package TribeEventsEnforcer
 */

if ( ! class_exists( 'Tribe_Events_Enforcer_Plugin_Update' ) ) {
	/**
	 * TribeEventsEnforcer Plugin Upgrader
	 */
	class Tribe_Events_Enforcer_Plugin_Update {

		/**
		 * Member Variable
		 *
		 * @var array $plugin_name Plugin name.
		 */
		protected $plugin_name;
		/**
		 * Member Variable
		 *
		 * @var string $current_version Current Plugin Version.
		 */
		protected $current_version;

		/**
		 * Member Variable
		 *
		 * @var string $slug Plugin Slug name.
		 */
		protected $slug;

		/**
		 * Member Variable
		 *
		 * @var string $plugin_file Plugin filename.
		 */
		protected $plugin_file;

		/**
		 * Member Variable
		 *
		 * @var string $transient_name Name of update transient.
		 */
		protected $transient_name = 'tribe_events_enforcer_update_plugins';

		/**
		 * Member Variable
		 *
		 * @var string $repo Plugin repo location.
		 */
		protected $repo = 'https://api.github.com/repos/Danny-Guzman/events-calendar-enforcer';

		/**
		 * Member Variable
		 *
		 * @var array $args Contains Header arguments used during the update process.
		 */
		protected $args;

		/**
		 * Initialize a new instance of the WordPress Auto-Update class
		 *
		 * @param string $plugin_slug Plugin slug name.
		 */
		public function __construct( $plugin_slug ) {
			$plugin_data = get_plugin_data( sprintf( '%1$s/%2$s/%2$s.php', WP_PLUGIN_DIR, $plugin_slug ) );

			// Set the class public variables.
			$this->current_version = $plugin_data['Version'];
			$this->plugin_name     = $plugin_data['Name'];
			$this->slug            = $plugin_slug;
			$this->plugin_file     = sprintf( '%1$s/%1$s.php', $plugin_slug );

			$this->args = array(
				'headers' => array(
					'Accept:'       => 'application/vnd.github.v3+json',
					'application/vnd.github.VERSION.raw',
					'application/octet-stream',
				),
			);

			// define the alternative API for plugin update checking.
			add_filter( 'pre_set_site_transient_update_plugins', array( $this, 'tribe_events_enforcer_check_plugin_update' ) );

			// Define the alternative response for information checking.
			add_filter( 'site_transient_update_plugins', array( $this, 'tribe_events_enforcer_add_plugins_to_update_notification' ) );

			add_filter( 'plugins_api', array( $this, 'tribe_events_enforcer_update_plugins_changelog' ), 20, 3 );

			// Define the alternative response for download_package which gets called during theme upgrade.
			add_filter( 'upgrader_pre_download', array( $this, 'download_package' ), 10, 3 );

			// Define the alternative response for upgrader_pre_install.
			add_filter( 'upgrader_source_selection', array( $this, 'tribe_events_enforcer_upgrader_source_selection' ), 10, 4 );

		}

		/**
		 * Alternative theme download for the WordPress Updater
		 *
		 * @see https://github.com/WordPress/WordPress/blob/master/wp-admin/includes/class-wp-upgrader.php#L265
		 *
		 * @param  bool        $reply Whether to bail without returning the package. Default false.
		 * @param  string      $package The package file name.
		 * @param  WP_Upgrader $upgrader The WP_Upgrader instance.
		 *
		 * @return string
		 */
		public function download_package( $reply, $package, $upgrader ) {
			if ( isset( $upgrader->skin->plugin_info ) && $upgrader->skin->plugin_info['Name'] === $this->plugin_name ) {
				$theme = wp_remote_retrieve_body( wp_remote_get( $package, array_merge( $this->args, array( 'timeout' => 60 ) ) ) );
				// Now use the standard PHP file functions.
				$fp = fopen( sprintf( '%1$s/%2$s.zip', plugin_dir_path( __DIR__ ), $this->slug ), 'w' );
				fwrite( $fp, $theme );
				fclose( $fp );

				return sprintf( '%1$s/%2$s.zip', plugin_dir_path( __DIR__ ), $this->slug );
			}
			return $reply;
		}

		/**
		 * Alternative API for checking for plugin updates.
		 *
		 * @param  array $update_transient Transient containing plugin updates.
		 *
		 * @return array
		 */
		public function tribe_events_enforcer_check_plugin_update( $update_transient ) {
			if ( ! isset( $update_transient->checked ) ) {
				return $update_transient;
			}

			$plugins = $update_transient->checked;

			$last_update = new stdClass();

			$payload = wp_remote_get( sprintf( '%1$s/releases/latest', $this->repo ), $this->args );

			if ( ! is_wp_error( $payload ) && wp_remote_retrieve_response_code( $payload ) === 200 ) {
				$payload = json_decode( wp_remote_retrieve_body( $payload ) );

				if ( ! empty( $payload ) && version_compare( $payload->tag_name, $this->current_version, '>' ) ) {
					$obj                 = new StdClass();
					$obj->name           = $this->plugin_name;
					$obj->slug           = $this->slug;
					$obj->plugin         = $this->plugin_file;
					$obj->new_version    = $payload->tag_name;
					$obj->published_date = ( new DateTime( $payload->published_at ) )->format( 'm/d/Y' );
					$obj->package        = $payload->zipball_url;
					$obj->tested         = '5.4.1';

					$theme_response = array( $this->plugin_file => $obj );

					$update_transient->response = array_merge( ! empty( $update_transient->response ) ? $update_transient->response : array(), $theme_response );

					$last_update->checked  = $plugins;
					$last_update->response = $theme_response;
				} else {
					delete_site_transient( $this->transient_name );
				}
			}

			$last_update->last_checked = time();
			set_site_transient( $this->transient_name, $last_update );

			return $update_transient;
		}

		/**
		 * Adds the EventsCalendarEnforce Plugin Update Notification to List of Available Updates.
		 *
		 * @param  array $update_transient Transient containing plugin updates.
		 *
		 * @return array
		 */
		public function tribe_events_enforcer_add_plugins_to_update_notification( $update_transient ) {
			$tribe_events_enforcer_update_plugins = get_site_transient( $this->transient_name );
			if ( ! is_object( $tribe_events_enforcer_update_plugins ) || ! isset( $tribe_events_enforcer_update_plugins->response ) ) {
				return $update_transient;
			}
			// Fix for warning messages on Dashboard / Updates page.
			if ( ! is_object( $update_transient ) ) {
				$update_transient = new stdClass();
			}

			$update_transient->response = array_merge(
				! empty( $update_transient->response ) ? $update_transient->response : array(),
				$tribe_events_enforcer_update_plugins->response
			);

			return $update_transient;
		}

		/**
		 * Filters the response for the current WordPress.org Plugin Installation API request.
		 * 
		 * @see https://developer.wordpress.org/reference/functions/plugins_api/
		 * 
		 * @param  false|object|array $result The result object or array. Default false.
		 * @param  string             $action The type of information being requested from the Plugin Installation API.
		 * @param  object             $args Plugin API arguments.
		 *
		 * @return false|object|array
		 */
		public function tribe_events_enforcer_update_plugins_changelog( $result, $action, $args ) {
			if ( isset( $args->slug ) && $args->slug === $this->slug ) {
				$tribe_events_enforcer_update_plugins = get_site_transient( $this->transient_name );
				if ( isset( $tribe_events_enforcer_update_plugins->response ) && isset( $tribe_events_enforcer_update_plugins->response[ $this->plugin_file ] ) ) {
					$tmp = $this->plugin_details();

					$tmp['version']      = $tribe_events_enforcer_update_plugins->response[ $this->plugin_file ]->new_version;
					$tmp['last_updated'] = $tribe_events_enforcer_update_plugins->response[ $this->plugin_file ]->published_date;

					$tmp['sections']['Changelog'] = $this->tribe_events_enforcer_get_plugin_changelog( $tmp['version'] );

					return (object) $tmp;
				}
			}
			return $result;
		}

		/**
		 * Retrieve Plugin Changelog from repository
		 *
		 * @param  string $ver Repository branch.
		 *
		 * @return string
		 */
		public function tribe_events_enforcer_get_plugin_changelog( $ver = 'master' ) {
			$logurl = sprintf( '%1$s/contents/changelog.txt?ref=%2$s', $this->repo, $ver );

			$changelog = wp_remote_get( $logurl, $this->args );

			if ( ! is_wp_error( $changelog ) && 200 === wp_remote_retrieve_response_code( $changelog ) ) {
				return sprintf( '<pre style="white-space: pre-line !important;">%1$s</pre>', base64_decode( json_decode( wp_remote_retrieve_body( $changelog ) )->content ) );
			} else {
				return 'No Changelog Available';
			}
		}

		/**
		 * Filters the source file location for the upgrade package.
		 *
		 * @see https://github.com/WordPress/WordPress/blob/master/wp-admin/includes/class-wp-upgrader.php#L524
		 *
		 * @param  string      $src File source location.
		 * @param  string      $rm_src Remote file source location.
		 * @param  WP_Upgrader $upgr WP_Upgrader instance.
		 * @param  array       $options Extra arguments passed to hooked filters.
		 *
		 * @return string
		 */
		public function tribe_events_enforcer_upgrader_source_selection( $src, $rm_src, $upgr, $options ) {

			if ( ! isset( $options['plugin'] ) || $options['plugin'] !== $this->plugin_file ) {
				return $src;
			}

			$tmp = explode( '/', $src );
			array_shift( $tmp );
			array_pop( $tmp );
			$tmp[ count( $tmp ) - 1 ] = $tmp[ count( $tmp ) - 2 ];
			$tmp                      = sprintf( '/%1$s/', implode( '/', $tmp ) );

			rename( $src, $tmp );

			return $tmp;
		}

		/**
		 * Plugin Details
		 * 
		 * @see https://developer.wordpress.org/reference/functions/plugins_api/
		 * @return array
		 */
		public function plugin_details() {
			$view_details = array(
				'slug'     => plugin_basename( plugin_dir_path( __DIR__ ) ),
				'author'   => 'Danny Guzman',
				'name'     => 'Force WP Template for The Events Calendar Plugin',
				'sections' => array(
					'Description' => 'Forces The Events Calendar Plugin to use the WordPress template hierarchy for events instead of its own templates.',
				),
				'requires' => '5.0.0',
				'tested' => '5.4.1',
			);

			return $view_details;
		}

	}
}

new Tribe_Events_Enforcer_Plugin_Update( plugin_basename( plugin_dir_path( __DIR__ ) ) );

